### PR TITLE
DOC-593 firstLoadingAnimation: fixed inconsistencies with Developers doc

### PR DESCRIPTION
**Pull request by fbeaudoincoveo**
- Also made some minor cosmetic changes to the SearchInterface documentation.

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/SearchInterface/SearchInterface.ts [hunk 1]
```diff
@@ -138,7 +138,7 @@ export class SearchInterface extends RootComponent implements IComponentBindings
      * > {@link Facet.options.dropdownHeaderLabel} and {@link Tab.options.dropdownHeaderLabel}).
      * >
      * > Furthermore, it is possible to specify the pixel threshold at which Facet components will go in responsive
-     * > mode (see {@link Facet.options.responsiveBreakpoint}.
+     * > mode (see {@link Facet.options.responsiveBreakpoint}).
```
Comment (olamothe) [inline src/ui/SearchInterface/SearchInterface.ts]: We modified that, but the doc has not been adjusted. 

The breakpoint is no longer specified per component. It's a global setting, on the search interface component (SearchInterface.responsiveComponents).

So this should be modified.

Comment (fbeaudoincoveo) [inline src/ui/SearchInterface/SearchInterface.ts]: Thanks! Should I also deprecate [Facet.options.responsiveBreakpoint](https://coveo.github.io/search-ui/components/facet.html#options.responsivebreakpoint) then?

Comment (olamothe) [inline src/ui/SearchInterface/SearchInterface.ts]: Yes

File: src/ui/Facet/Facet.ts [hunk 2]
```diff
@@ -604,8 +604,11 @@ export class Facet extends Component {
      * framework uses the last occurrence of the option.
      *
      * Default value is `800`.
+     *
+     * @deprecated This option is exposed for legacy reasons, and the recommendation is to not use this option. See
+     * {@link SearchInterface.responsiveComponents} instead.
      */
-    responsiveBreakpoint: ComponentOptions.buildNumberOption({ defaultValue: 800 }),
+    responsiveBreakpoint: ComponentOptions.buildNumberOption({ defaultValue: 800, deprecated: 'This option is exposed for legacy reasons, and the recommendation is to not use this option.' }),
```
Comment (fbeaudoincoveo) [inline src/ui/Facet/Facet.ts]: Yes, I did exactly this in the previous commit :)

Comment (olamothe) [inline src/ui/Facet/Facet.ts]: I would also go as far as completely removing the JSDOC for deprecated doc. It serves pretty much no purposes, and only pollutes the doc website with useless info.

Comment (fbeaudoincoveo) [inline src/ui/Facet/Facet.ts]: I just realized I understood what you said the other way around ;)

I'll make a new commit to fix this.

**Pull request merged**